test/SlowTests/Issues/RavenDB-17650.cs

```diff
@@ -0,0 +1,236 @@
+            }
+
+            await store.Subscriptions
+                .CreateAsync(new SubscriptionCreationOptions<User>
+                {
+                    Name = "BackgroundSubscriptionWorker"
+                });
+            
+            using var worker = store.Subscriptions.GetSubscriptionWorker<User>(new SubscriptionWorkerOptions("BackgroundSubscriptionWorker"));
+
+            // disable database
+            var disableSucceeded = store.Maintenance.Server.Send(new ToggleDatabasesStateOperation(store.Database, true));
+            Assert.True(disableSucceeded.Success);
+            Assert.True(disableSucceeded.Disabled);
+
+            var cts = new CancellationTokenSource();
+            var failMre = new ManualResetEvent(false);
+            worker.OnSubscriptionConnectionRetry += _ =>
+            {
+                failMre.Set();
+            };
+            var successMre = new ManualResetEvent(false);
+            var _ = worker.Run( batch =>
+            {
+                successMre.Set();
+            }, cts.Token);
+
+            //enable database
+            Assert.True(failMre.WaitOne(TimeSpan.FromSeconds(15)), "Subscription didn't fail as expected.");
+            var enableSucceeded = store.Maintenance.Server.Send(new ToggleDatabasesStateOperation(store.Database, false));
+            Assert.False(enableSucceeded.Disabled);
+            Assert.True(enableSucceeded.Success);
+            Assert.True(successMre.WaitOne(TimeSpan.FromSeconds(15)), "Subscription didn't success as expected.");
+        }
+
+        [Fact]
+        public async Task Should_Retry_When_AllTopologyNodesDownException_Was_Thrown()
+        {
+            var node = GetNewServer(new ServerCreationOptions { RunInMemory = false });
+            using var store = GetDocumentStore(new Options()
+            {
+                ReplicationFactor = 1,
+                RunInMemory = false,
+                Server = node
+            });
+            string id = "User/33-A";
+            using (var session = store.OpenAsyncSession())
+            {
+                await session.StoreAsync(new User { Id = id, Name = "1" });
+                await session.StoreAsync(new User { Name = "2" });
+                await session.SaveChangesAsync();
+            }
+
+            await store.Subscriptions
+                .CreateAsync(new SubscriptionCreationOptions<User>
+                {
+                    Name = "BackgroundSubscriptionWorker"
+                });
+
+            using var worker = store.Subscriptions.GetSubscriptionWorker<User>(new SubscriptionWorkerOptions("BackgroundSubscriptionWorker"));
+
+            // dispose nodes
+            var result = await DisposeServerAndWaitForFinishOfDisposalAsync(node);
+
+            var cts = new CancellationTokenSource();
+            var failMre = new ManualResetEvent(false);
+            worker.OnSubscriptionConnectionRetry += _ =>
+            {
+                failMre.Set();
```

Comment:
lets check that we actually get `AllTopologyNodesDownException` here

Reply:
Done